dojo/tools/ossindex_devaudit/parser.py

```diff
@@ -0,0 +1,98 @@
+import json
+
+from dojo.models import Finding
+
+
+class OssIndexDevauditParser(object):
+    def __init__(self, json_file, test):
+
+        tree = self.parse_json(json_file)
+
+        if tree:
+            self.items = [data for data in self.get_items(tree, test)]
+        else:
+            self.items = []
+
+    def parse_json(self, json_file):
+        if json_file is None:
+            self.items = []
+            return
+        try:
+            tree = json.load(json_file)
+        except:
+            raise Exception("Invalid format")
+
+        return tree
+
+    def get_items(self, tree, test):
+
+        items = {}
+
+        results = {key: value for (key, value) in tree.items()}
+        for package in results['Packages']:
+            package_data = package['Package']
+            if len(package['Vulnerabilities']) > 0:
+                for vulnerability in package['Vulnerabilities']:
+                    item = get_item(
+                        dependency_name=package_data['name'],
+                        dependency_version=package_data['version'],
+                        dependency_source=package_data['pm'],
+                        vulnerability=vulnerability,
+                        test=test
+                    )
+                    unique_key = vulnerability['id']
+                    items[unique_key] = item
+
+        return items.values()
+
+
+def get_item(dependency_name, dependency_version, dependency_source, vulnerability, test):
+
+    if vulnerability['cwe'] is None:
+        cwe_text = ''
+        cwe = 1035
+    else:
+        cwe_text = vulnerability['cwe']
+        cwe = vulnerability['cwe'].replace('CWE-', '')

```

Comment:
cwe is an integer field so you should check it's an integer. if not, assign 1035.

Reply:
Cwe handling refactored